pkg/resiliency/resiliency.go

```diff
@@ -827,18 +827,21 @@ func (r *Resiliency) GetPolicy(target string, policyType PolicyType) *PolicyDesc
 		componentPolicy, exists = r.components[target]
 		if exists {
 			policy, _ := policyType.(*ComponentPolicy)
-			if policy.componentDirection == "Inbound" {
+			switch policy.componentDirection {
+			case "Inbound":
 				policyName = PolicyNames{
 					Retry:          componentPolicy.Inbound.Retry,
 					CircuitBreaker: componentPolicy.Inbound.CircuitBreaker,
 					Timeout:        componentPolicy.Inbound.Timeout,
 				}
-			} else {
+			case "Outbound":
 				policyName = PolicyNames{
 					Retry:          componentPolicy.Outbound.Retry,
 					CircuitBreaker: componentPolicy.Outbound.CircuitBreaker,
 					Timeout:        componentPolicy.Outbound.Timeout,
 				}
+			default:
+				panic(fmt.Errorf("invalid component policy direction: '%s'", policy.componentDirection))
```

Comment:
I think we should not panic here. We should return an error and handle that error correctly.

Reply:
IMHO a panic here is appropriate because this should never happen during runtime. It's only here to help catch development-time errors.

Reply:
my question is should the application crash because of an invalid resiliency policy? Should it not gracefully degrade and use default polices where applicable and let user know that there is an error?

In this case, my opinion is that a faulty resiliency policy config should not cause the application to crash.

Reply:
Oh an invalid resiliency policy would not cause this crash. Only using the wrong constant in the source code would.

Reply:
went back and saw the full code flow for reading in resiliency policy. Agree with your comment.